src/Render/Render.jl

```diff
@@ -7,4 +7,33 @@ Render
 
 using Compat
 
+import ..Elements: Node
+
+
+immutable EmptyRenderError <: Exception
+    msg :: AbstractString
+end
+
+function checkchildren(n::Node)
+    isempty(n.children) && throw(EmptyRenderError(string("'$(typeof(n))': '",
+                                haskey(n.data, :title) ?  n.data[:title] : n.data[:outname],
+                                "' -> has no content")))
+end
+
+function getoutname(n::Node)
+    haskey(n.data, :outname) && return n.data[:outname]
+    if haskey(n.data, :title)
+        replace_chars = Set(Char[' ', '&', '-'])
+        io = IOBuffer()
+        for c in n.data[:title]
+            c in replace_chars ? write(io, "_") : write(io, lowercase(string(c)))
+        end
+        return takebuf_string(io)
+    end
```

Comment:
Since this method might get called quite often for the same node, it would be good to cache the generated ``:outname`` before returning it so it only has to be done once.